### ISSUE TITLE
Custom Script CLI and API execution not working

### ISSUE THREAD

**Issue by wz4** [netbox-community/netbox#17923]
### Deployment Type

Self-hosted

### Triage priority

N/A

### NetBox Version

v4.1.6

### Python Version

3.11

### Steps to Reproduce

Unable to execute custom scripts via the CLI and API. This is the same issue referenced in #17885. The fix implemented only applies to the GUI. 

Adding the script name as an argument to the enqueue method fixes the problem.

https://github.com/netbox-community/netbox/blob/27d15615b3590000901984bddfa7a815cd1343d8/netbox/extras/api/views.py#L281-L290

https://github.com/netbox-community/netbox/blob/27d15615b3590000901984bddfa7a815cd1343d8/netbox/extras/management/commands/runscript.py#L84-L100

### Expected Behavior

Executing a script for the CLI and API should work

### Observed Behavior

The following error is produced when attempting to use the CLI runscript command
```
  File "/opt/netbox/netbox/extras/management/commands/runscript.py", line 85, in handle
    job = ScriptJob.enqueue(
          ^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/netbox/netbox/jobs.py", line 106, in enqueue
    return Job.enqueue(cls.handle, name=name, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/netbox/core/models/jobs.py", line 236, in enqueue
    job.full_clean()
  File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/models/base.py", line 1561, in full_clean
    raise ValidationError(errors)
django.core.exceptions.ValidationError: {'name': ['This field cannot be blank.']}
```

**Comment by cruse1977** [comment]
+1 to this

**Comment by alehaa** [comment]
I repeat my comment from #17885 and volunteer for submitting a PR:

> The bug originates from the following code. The justification of this was to not have the script name twice (as job name and related object) listed in the job table. Just having "SciptJob" as generic name seemed a bit odd to me at that time. https://github.com/netbox-community/netbox/blob/58bc3884578a98595b02d7d7412c83d41fc7bbfa/netbox/extras/jobs.py#L24-L27
> […] Reverting #17886 and removing the code linked above should fix the issue and avoid confusion. `ScriptJob` will be used as default name afterwards and the script linked as job object.

**Comment by arthanson** [comment]
@alehaa  do you want to submit a PR, can you also check if it addresses #17921?

**Comment by llamafilm** [comment]
@alehaa are you still planning on making a PR?

**Comment by alehaa** [comment]
There is already one linked to this issue. :)

**Comment by llamafilm** [comment]
Thanks!  I missed that.  I tested your change now and I can confirm it fixes the problem for both CLI and scheduled script execution.

**Comment by jeremystretch** [comment]
I believe this has been resolved by PR #17932.
